src/input.rs

```diff
@@ -72,13 +73,35 @@ impl Input {
         self
     }
 
-    /// Sets a validation callback for the input.
+    /// Sets a validation callback for the input that is called when the user submits.
+    /// The same as [`Input::validate_on_enter`].
     pub fn validate<V>(mut self, validator: V) -> Self
     where
         V: Validate<String> + 'static,
         V::Err: ToString,
     {
-        self.validate = Some(Box::new(move |input: &String| {
+        self.validate_on_enter = Some(Box::new(move |input: &String| {
```

Comment:
Given the naming convention here, I was expecting `validate()` to be deprecated and be a shim that calls out to `validate_on_enter()` as opposed to the other way around. Doesn't really matter functionality-wise, but just thought I'd raise the point.

Reply:
I mean, the original idea was to keep `cliclack` close to `@clack/prompts`, so let it be without deprecation, it doesn't make any harm anyways.